### PR TITLE
Social media block

### PR DESCRIPTION
**Pull request by davidpmccormick**
References #668.

## Type
✨ Feature  

## Value
Abstracts the 'tweet-block' component into a 'social-media-block' for use with Instagram.

## Screenshot
screen shot 2017-06-07 at 12 07 47

## TODO
- [x] Update social-media-block config to use instagram-thumbnail

### All
- [x] Demoed to the relevant people
- [x] PR labelled and assigned
- [x] Any introduced code complexity has been flagged

### UI component only
- [x] A11y tested: `npm run test:accessibility <URL>`
- [x] Browser tested: `npm run test:browsers <URL>`
- [x] Works without JS in the client ('works' = doesn't display)
- [x] Relevant tracking/monitoring has been considered

### CODE REVIEW

File: server/controllers/index.js [hunk 1]
```diff
@@ -208,8 +123,7 @@ export const explore = async(ctx, next) => {
     topPromo,
     second3Promos,
     next8Promos,
-    collectorsPromo,
-    latestTweets
```
Comment (davidpmccormick) [inline server/controllers/index.js]: I believe this was the unnecessary offending synchronous request for latest tweets that has caused page load to increase.

File: server/views/components/social-media-block/index.njk [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      </div>
+      <a href="{{ model.url }}"
+        class="social-media-block__handle flex flex--v-center flex--h-center {{ {s:6, m:6, l:6} | spacingClasses({margin: ['bottom']}) }}">
+        @{{ model.handle }}
+      </a>
+      <div class="grid {{ {s:10, m:10, l:10} | spacingClasses({padding: ['bottom']}) }}">
+        {% if model.service === 'Twitter' %}
+          {% for post in model.posts %}
+            <div class="{{ {s:4, m:4, l:6, xl:3} | gridClasses }}">
+              <div data-track-click='{"screenName": "{{ post.screenName }}", "createdAt": "{{ post.createdAt }}", "postId": {{ post.id }}}'>
```
Comment (jamesgorrie) [inline server/views/components/social-media-block/index.njk]: Again, wondering if we actually want all this?
Just wondering why this is outside of the component and the Instagram one is contained in it?

File: server/controllers/async-controllers.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+  ctx.body = {
+    html: ctx.body
+  };
+
+  return next();
+};
+
+export const latestTweets = async(ctx, next) => {
+  const count = ctx.params.count || 4;
```
Comment (jamesgorrie) [inline server/controllers/async-controllers.js]: We never set this anywhere do we?

File: client/scss/components/_instagram_thumbnail.scss [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  display: block;
+  position: relative;
+
+  .icon {
+    width: 1em;
+    height: 1em;
+    margin-right: 0.2em;
+  }
+
+  .icon__shape {
```
Comment (davidpmccormick) [inline client/scss/components/_instagram_thumbnail.scss]: A class per colour in our palette? I could get with that, yep.

File: server/controllers/async-controllers.js (hunk 3 above)
Comment (davidpmccormick) [inline server/controllers/async-controllers.js]: Not yet, no – do you think it's YAGNI to have it as a parameter?

File: server/views/components/instagram-thumbnail/index.njk [hunk 5]
```diff
@@ -0,0 +1,31 @@
+<a href="{{ model.url }}" class="instagram-thumbnail"
+  data-track-click='{"screenName": "{{ model.screenName }}", "createdAt": "{{ model.createdAt }}", "postId": {{ model.id }}}'>
```
Comment (davidpmccormick) [inline server/views/components/instagram-thumbnail/index.njk]: I think they all have some merit/can't harm – it's a relatively straightforward step to filter out noise in Mixpanel.

Comment (jamesgorrie) [inline server/views/components/instagram-thumbnail/index.njk]: > Can't harm
Adding things for people to have to filter through, irrespective of how simple it is, seems unnecessary?

But if we can see merit, that's alright, but could we perhaps hypothesise why this data might be useful?

Isn't the `screenName` always going to be ours, and `createdAt` just be the latest posts, making it useless to aggregate or filter through?

File: server/controllers/async-controllers.js (hunk 3 above)
Comment (jamesgorrie) [inline server/controllers/async-controllers.js]: Indeed - let's add things when we need them - the code base is getting bigger, and more complicated.

If I saw this, I would assume we're using it, so not change it if I needed and potentially work around it, making refactoring harder to fathom.

File: client/scss/components/_instagram_thumbnail.scss (hunk 4 above)
Comment (jamesgorrie) [inline client/scss/components/_instagram_thumbnail.scss]: yea, is doesn't like we have many icon shades... Another PR thought,

File: server/views/components/instagram-thumbnail/index.njk (hunk 5 above)
Comment (davidpmccormick) [inline server/views/components/instagram-thumbnail/index.njk]: `screenName` could be a retweet. The tweet's creation date seemed like quite a useful bit of information to hold on to to me, but I won't fight too hard for it to stay.

Comment (jamesgorrie) [inline server/views/components/instagram-thumbnail/index.njk]: Analytics for analytics sake is something GA is really good at.
Mixpanel is really good at hyper focussed analytics.

Maybe speak to @hthair @jennpb and @Heesoomoon about what they'd like?

File: client/scss/components/_instagram_thumbnail.scss [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  z-index: 1;
+}
+
+.instagram-thumbnail__type-text {
+  @include visually-hidden;
+}
+
+.instagram-thumbnail__overlay {
+  position: absolute;
+  top: 0;
```
Comment (jamesgorrie) [inline client/scss/components/_instagram_thumbnail.scss]: What does this do?

Comment (davidpmccormick) [inline client/scss/components/_instagram_thumbnail.scss]: Aligns the semi-opaque overlay over the image

screen shot 2017-06-09 at 15 53 29

**Comment by Norvard** [comment]
Ah yes I forgot that people can post all sorts of image ratios these days...
@davidpmccormick Can we make all those images bottom align? Will be in line with other such layouts we have

**Comment by davidpmccormick** [comment]
> Can we make all those images bottom align?

We can. We spoke. You didn't like how it looked. Will keep it as is.

**Pull request merged**